zetaclient/chains/bitcoin/observer/witness.go

```diff
@@ -47,11 +48,14 @@ func GetBtcEventWithWitness(
 		return nil, errors.Wrapf(err, "error calculating depositor fee for inbound %s", tx.Txid)
 	}
 
-	isAmountValid, amount := isValidAmount(tx.Vout[0].Value, depositorFee)
-	if !isAmountValid {
-		logger.Info().
-			Msgf("GetBtcEventWithWitness: btc deposit amount %v in txid %s is less than depositor fee %v", tx.Vout[0].Value, tx.Txid, depositorFee)
-		return nil, nil
+	// deduct depositor fee
+	// to allow developers to track failed deposit caused by insufficient depositor fee,
+	// the error message will be forwarded to zetacore to register a failed CCTX
+	status := types.InboundStatus_SUCCESS
+	amount, err := DeductDepositorFee(tx.Vout[0].Value, depositorFee)
+	if err != nil {
+		status = types.InboundStatus_INSUFFICIENT_DEPOSITOR_FEE
```

Comment:
Can we set amount to 0? We should use amount if err is returned in the function

Reply:
Yeah, the function `DeductDepositorFee` returns 0 `amount` when error happens.
```
if deposited < depositorFee {
    return 0, fmt.Errorf("deposited amount %v is less than depositor fee %v", deposited, depositorFee)
}
```

Reply:
Even so, if the function return `err != nil` it is cleaner to consider other returned values as undefined

Reply:
I set it to `0` explicitly in commit https://github.com/zeta-chain/node/pull/3358/commits/7040696a6dc46e278599ef9d0721cd825d96cca9.